### DISCUSSION TITLE
Large-scale full-stack Nuxt architecture

### DISCUSSION THREAD

**Discussion by hymair** [nuxt/nuxt#27376] [Questions]
We have a pretty large Nest+Nuxt combo within a monorepo, but with the advancements made in Nuxt 3, we think we might be able to switch to Nuxt fully and move all our backend code to the `server/` dir. This will save us all the headache of building multiple apps within the monorepo (app, backend, UI, types packages).

We think the Nest style of organising the projects into modules is nice. But we also like the file-based routing system proposed via Nitro. File-based routing means that the controllers would be within `/server/api`, but where do the services fit then? We have quite a bit of business logic, both shared and specific to the module/route.

Any ideas? Anyone built anything large scale like this using Nuxt full-stack? (we have 30 nest modules currently)

**Comment by AloisSeckar** [comment]
I didn't, but I would propose simply `/server/modules/<module>/*` to wrap and preserve file/folder structure of your modules.

Btw. you don't necessary need to place controlers under `/server/api`. You can as well use `/server/routes` - this allows fully custom routes without having to include "/api" eveywhere. Folder `/server/api` is effectively a shorthand for `/server/routes/api`.

**Comment by TheAlexLichter** [comment]
I'd go for multiple [Nuxt layers](https://www.youtube.com/watch?v=fr5yo3aVkfA) or modules (depending on your need :) )
